### PR TITLE
Replace 'request.json' by 'request.get_json' if flask is greater than…

### PR DESCRIPTION
**Pull request by dhilst**
… 0.12.2.

Request.json is deprecated since Flask 0.12.2. This commit fix
the handling of this field avoiding a warning. The warning only happens
over Python 3.

Tested with python 3.5.3

### CODE REVIEW

**Comment by dhilst** [comment]
The warning is only triggered with python3. Here is how to test it:

https://gist.github.com/dhilst/6e4e76824467d21f12b74f1bd53dbf35

**Comment by dhilst** [comment]
I need an opinion. The first commit try to keep compatibility with an ugly diff. The
second do more conservative changes. What you guy prefer?

The third commit fixes the tests.

**Comment by dhilst** [comment]
screenshot from 2017-06-02 19-17-54

Here is an example of the warning.

**Comment by donato** [comment]
We are seeing these same warnings at our company as well (JW Player), looking forward to the cleanup!

File: flask_restful/reqparse.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def __setattr__(self, name, value):
         self[name] = value
 
 _friendly_location = {
-    u'json': u'the JSON body',
+    u'get_json': u'the JSON body',
```
[the remark this answers is not in the document]
Comment (dhilst) [reply flask_restful/reqparse.py]: Fixed!

**Comment by galuszkak** [comment]
@dhilst this isn't a case anymore. With Flask 1.0. flask `.json` isn't deprecated anymore.

**Comment by dhilst** [comment]
@galuszkak No problem! :-)

**Comment by joshfriend** [comment]
I guess the issue solved itself! Thank you for the PR though :)

**Pull request closed without merging**